waspc/data/Generator/templates/server/src/auth/providers/oauth/createRouter.ts

```diff
@@ -49,40 +50,45 @@ export function createRouter(provider: ProviderConfig, initData: { passportStrat
           const getUserFields = () => getUserFieldsFn ? getUserFieldsFn(contextWithUserEntity, { profile: providerProfile }) : Promise.resolve({});
           // TODO: In the future we could make this configurable, possibly associating an external account
           // with the currently logged in account, or by some DB lookup.
-          const auth = await findOrCreateAuthByAuthIdentity(provider.id, providerProfile.id, getUserFields);
+          const userId = await getOrCreateUserIdByOAuthProvider(provider.id, providerProfile.id, getUserFields);
+
+          const token = await createAuthToken(userId);
 
-          const token = await sign(auth.{= userFieldOnAuthEntityName =}.id);
           res.json({ token });
       })
     )
 
     return router;
 }
 
-async function findOrCreateAuthByAuthIdentity(
+async function getOrCreateUserIdByOAuthProvider(
   providerName: string,
   providerUserId: string,
   getUserFields: () => ReturnType<GetUserFieldsFn>,
 ) {
-  // Attempt to find a User by an external auth association.
-  const authIdentity = await prisma.{= authIdentityEntityLower =}.findFirst({
-    where: { providerName, providerUserId },
-    include: {
-      {= authFieldOnAuthIdentityEntityName =}: {
-        include: {
-          {= userFieldOnAuthEntityName =}: true
+  try {
+    const authIdentity = await prisma.{= authIdentityEntityLower =}.findFirst({
+      where: { providerName, providerUserId },
+      include: {
+        {= authFieldOnAuthIdentityEntityName =}: {
+          include: {
+            {= userFieldOnAuthEntityName =}: true
+          }
         }
       }
+    })
+
+    
+    if (authIdentity) {
+      return authIdentity.{= authFieldOnAuthIdentityEntityName =}.{= userFieldOnAuthEntityName =}.id
     }
-  })
+  
+    const userFields = await getUserFields()
+  
+    const auth = await createAuthWithUser(providerName, providerUserId, undefined, userFields)
 
-  if (authIdentity) {
-    return authIdentity.{= authFieldOnAuthIdentityEntityName =}
+    return auth.userId;
+  } catch (e) {
+    rethrowPossibleAuthError(e)
```

Comment:
That's one pretty wide catch. What if error is not possible auth error, then it will just rethrow it as it is?

Reply:
Yes, if it finds something specific that it knows how to handle it will, else, it throws an `500` error.

Reply:
I hope else it rethrows the errors, and doesn't immediately make it 500 -> 500 should happen later by our default middleware?

Reply:
That was the better way to do it 👍 I've now changed the code to match your suggestion